bundles/org.eclipse.jface/src/org/eclipse/jface/dialogs/DialogSettings.java

```diff
@@ -364,6 +368,30 @@ public void put(String key, String value) {
 		}
 	}
 
+	public void inputNewSearchFilter(String SearchIn) {
+		lastFiveSearches.addFirst(SearchIn);
+
+		int index = -1;
+		for (int i = 1; i < lastFiveSearches.size(); i++) {
+			if (lastFiveSearches.get(i).equals(SearchIn)) {
+				index = i;
+			}
+		}
+		if (index != -1)
+			lastFiveSearches.remove(index);
+
```

Comment:
Isn't this equivalent to?
```suggestion
		lastFiveSearches.remove(searchIn);
```
And btw. accessing a `LinkedList` by index has a runtime that's linear to the list size, unlike `ArrayList` that has constant time and since you are doing it in a loop over all elements, this search has O(n^2) runtime.

Reply:
Tanks for the hint, I did not have in mind that if the element does not exist, nothing is removed. I thought that an error would be thrown. But i looked it up and it actually is equivalent